File: ios/ExtremeBeatAnnihilation/ExtremeBeatAnnihilation/SoundEvent.m
//
//  SoundEvent.m
//  ExtremeBeatAnnihilation
//
//  Created by Andrew Krepps on 11/2/13.
//  Copyright (c) 2013 Crash Course Gaming. All rights reserved.
//

#import "SoundEvent.h"

@implementation SoundEvent

-(id) initWithSample:(uint)s andFreq:(float)f
{
    if (self = [super init])
    {
        _sample = s;
        _freq = f;
    }
    
    return self;
}

-(void) setAttributesWithQ1:(float)q1Freq andMedian:(float)medianFreq andQ3:(float)q3Freq
{
    if (_freq <= q1Freq)
    {
        _side = LEFT;
        _type = JUMP;
    }
    else if (self.freq <= medianFreq)
    {
        _side = LEFT;
        _type = JUMP;
    }
    else if (self.freq <= q3Freq)
    {
        _side = RIGHT;
        _type = SLIDE;
    }
    else
    {
        _side = RIGHT;
        _type = JUMP;
    }
}

@end

File: android/ExtremeBeatAnnihilation/Classes/SoundEvent.cpp
#include "SoundEvent.h"

using namespace cocos2d;

// needs description
//
// s [in] - ?
// f [in] - ?
void SoundEvent::InitWithSample(unsigned int s, float f)
{
    m_sample = s;
    m_freq   = f;
}

// needs description
//
// q1Freq     [in] - ?
// medianFreq [in] - ?
// q3Freq     [in] - ?
void SoundEvent::SetAttributesWithQ1(float q1Freq, float medianFreq, float q3Freq)
{
    if (m_freq <= q1Freq)                // needs description
    {
        m_side = LEFT;
        m_type = SLIDE;
    }
    else if (m_freq <= medianFreq)       // needs description
    {
        m_side = LEFT;
        m_type = JUMP;
    }
    else if (m_freq <= q3Freq)           // needs description
    {
        m_side = RIGHT;
        m_type = SLIDE;
    }
    else                                 // needs description
    {
        m_side = RIGHT;
        m_type = JUMP;
    }
}

float SoundEvent::GetFreq(void)
{
	return m_freq;
}
